nwb_conversion_tools/datainterfaces/ecephys/intan/intandatainterface.py

```diff
@@ -53,6 +43,29 @@ def get_metadata(self):
                 channel_id=channel_id, property_name="group_name", value=f"Group{channel_group}"
             )
 
+        if len(unique_group_names) > 1:
+            for channel_id, group_electrode_number in zip(channel_ids, group_electrode_numbers):
+                self.recording_extractor.set_channel_property(
+                    channel_id=channel_id, property_name="group_electrode_number", value=group_electrode_number
+                )
+
+        custom_names = [channel["custom_channel_name"] for channel in valid_channels]
+        if any(custom_names):
+            for channel_id, custom_name in zip(channel_ids, custom_names):
+                self.recording_extractor.set_channel_property(
+                    channel_id=channel_id, property_name="custom_channel_name", value=custom_name
+                )
```

Comment:
@CodyCBakerPhD would it be possible to move these set operations to the `__init__`?